Bigtable/src/BigtableClient.php

```diff
@@ -141,4 +149,27 @@ public function table($instanceId, $tableId, array $options = [])
             $options
         );
     }
+
+    /**
+     * Attempts to detect the project ID.
+     *
+     * @todo Add better support for detecting the project ID (check keyFile/GCE metadata server).
+     * @return string
+     * @throws ValidationException If a project ID cannot be detected.
+     */
+    private function detectProjectId()
+    {
+        if (getenv('GOOGLE_CLOUD_PROJECT')) {
+            return getenv('GOOGLE_CLOUD_PROJECT');
+        }
+
+        if (getenv('GCLOUD_PROJECT')) {
+            return getenv('GCLOUD_PROJECT');
+        }
+
+        throw new ValidationException(
```

Comment:
what would happen if user provides just credential file? Shouldnt we move this code in some core package as this would be used by all client I think.

Reply:
After discussing this a bit offline we're going to move forward with the changes in this PR.